service/src/main/scala/za/co/absa/logingw/rest/config/BaseConfig.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2023 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.logingw.rest.config
+
+import org.springframework.boot.context.properties.{ConfigurationProperties, ConstructorBinding}
+
+@ConstructorBinding
+@ConfigurationProperties(prefix = "logingw.rest.config")
+class BaseConfig(
+                val someKey: String,
+                val algName: String,
+                val expTime: Int
+                )
```

Comment:
```suggestion
class BaseConfig(
  val algName: String,
  val expTime: Int,
  val someKey: String = "example"
)
```

Please adjust the indentation.
Suggesting someKey to go last (can be more easily dropped later) and have a default value (does not have to be set where not needed).